Public/Src/Cache/MemoizationStore/Library/Stores/RocksDbMemoizationStore.cs

```diff
@@ -0,0 +1,194 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+extern alias Async;
+
+using System;
+using System.Collections.Generic;
+using System.Diagnostics.ContractsLight;
+using System.Threading;
+using System.Threading.Tasks;
+using BuildXL.Cache.ContentStore.Distributed.NuCache;
+using BuildXL.Cache.ContentStore.Interfaces.Extensions;
+using BuildXL.Cache.ContentStore.Interfaces.Logging;
+using BuildXL.Cache.ContentStore.Interfaces.Results;
+using BuildXL.Cache.ContentStore.Interfaces.Sessions;
+using BuildXL.Cache.ContentStore.Interfaces.Time;
+using BuildXL.Cache.ContentStore.Interfaces.Tracing;
+using BuildXL.Cache.ContentStore.Tracing.Internal;
+using BuildXL.Cache.MemoizationStore.Interfaces.Results;
+using BuildXL.Cache.MemoizationStore.Interfaces.Sessions;
+using BuildXL.Cache.MemoizationStore.Interfaces.Stores;
+using BuildXL.Cache.MemoizationStore.Sessions;
+using BuildXL.Cache.MemoizationStore.Tracing;
+
+namespace BuildXL.Cache.MemoizationStore.Stores
+{
+    /// <summary>
+    ///     An IMemoizationStore implementation using RocksDb.
+    /// </summary>
+    public class RocksDbMemoizationStore : IMemoizationStore

```

Comment:
I think it make sense to derive from StartupShutdown types to simplify the implementation.